### PR TITLE
Utf8 encoding

### PR DESCRIPTION
**Pull request by jedwards4b**
Use the encoding flag to open in py3 to set utf-8 

Test suite: scripts_regression_tests.py py2 and py3
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes cdash testing with py3

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/lib/CIME/XML/generic_xml.py [hunk 1]
```diff
@@ -46,12 +46,20 @@ def read(self, infile, schema=None):
         Read and parse an xml file into the object
         """
         logger.debug("read: " + infile)
-        with open(infile, 'r') as fd:
-            if self.tree:
-                self.root.append(ET.parse(fd).getroot())
-            else:
-                self.tree = ET.parse(fd)
-                self.root = self.tree.getroot()
+        if six.PY3:
```
Comment (jgfouca) [inline scripts/lib/CIME/XML/generic_xml.py]: Nice catch for this bug. Did you figure out why it worked from the command line? You were still using python3 on the command-line, right?

Anyway, I think we can reduce some code duplication here, even though py2 does not support encoding:
```
file_open = (lambda x: open(x, 'r', encoding='utf-8')) if six.PY3 else (lambda x: open(x, 'r'))
with file_open(infile) as fd:
    ...
```

Comment (jedwards4b) [reply scripts/lib/CIME/XML/generic_xml.py]: Thanks - I thought there must be a way to do that.  Do you want to push that change?

Comment (jedwards4b) [reply scripts/lib/CIME/XML/generic_xml.py]: I don't understand why it worked from the command line but not from cron.

**Comment by jedwards4b** [comment]
Okay, i've pushed the suggested change

**Pull request merged**